include/libtorrent/aux_/session_impl.hpp

```diff
@@ -1054,15 +1065,28 @@ namespace aux {
 				, error_code& ec
 				, int flags);
 
+			void send_udp_packet_hostname_listen(aux::session_listen_socket* sock
```

Comment:
my suspicion is that each udp socket will contact the socks5 proxy from its interface and try to set up a UDP associate tunnel, and presumably only the first one will succeed, as it tries to bind to a specific port. so, whatever the other ones do, probably doesn't matter. Maybe at some point someone will want to improve this, but I think this is fine for now.